RCNN_tf.py

```diff
@@ -116,209 +176,292 @@ def loop_body(x, recurrent_states, output):
 
         return output
 
-
-def convolutional_layer(
-        input_data,
-        filter_shape,
-        num_input_channels,
-        num_filter,
-        std=.03,
-        stride=(1, 1),
-        padding='same',
-        name='conv'
-):
-    conv_filter_shape = [filter_shape[0], filter_shape[1], num_input_channels, num_filter]
+    else:
+        raise Exception("Processing unit has to be one of: cpu, gpu.")
+
+
+def convolutional_layer(input_data, num_input_chans, num_filter, filter_shape,
+        stddev=.03, stride=(1, 1), padding='same', name='conv'):
+    """Defines a convolutional layer
+
+    Args:
+        input_data   (tf.Tensor): 4D data tensor
+        num_input_chans    (int): the number of input channels
+        num_filter         (int): the number of filters
+        filter_shape      (list): the shape of the 2D filter
+        stddev           (float): standard deviation
+        stride           (tuple): ???
+        padding            (str): ???
```

Comment:
I can add the docstrings. For further information about stride and padding:
https://www.quora.com/What-are-the-roles-of-stride-and-padding-in-a-convolutional-neural-network

Reply:
Ah, yeah: I know what these mean haha.
I just forgot to come back to it, my bad.